### PR TITLE
Contribution guide

### PR DESCRIPTION
**Pull request by radiocontrolled**
Resolves https://github.com/BBC-News/psammead/issues/36

Guidelines for contributing to Psammead.

- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+```
+├── README.md
+├── CHANGELOG.md
+├── __snapshots__
+│   └── index.test.jsx.snap
+├── index.jsx
+├── index.stories.jsx
+├── index.test.jsx
+└── package.json
```
Comment (bcmn) [inline CONTRIBUTING.md]:
```suggestion
├── package.json
└── package-lock.json
```

**Review by bcmn — DISMISSED** [review]
Looks good! Small suggestion, maybe it's already been considered.

**Review by bcmn — DISMISSED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+```
+- ## Description (high-level summary that will be accessible to product/UX/Test, plus link to Storybook, explain the different cases)
+- ## When to use this component (including where it is currently used)
+- ## When to not use this component (if this is not known, can leave the section empty)
+- ## Accessibility notes (ensuring they make sense to non-technical audiences)
+- ## Roadmap (what we have planned for this component)
+- ## Additional notes (link to any relevant ADRs)
+```
+
+In addition, the `README.md` should also point to the main Psammead [CODE_OF_CONDUCT](https://github.com/BBC-News/psammead/blob/latest/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/BBC-News/psammead/blob/latest/CONTRIBUTING.md) files.
```
Comment (dr3) [inline CONTRIBUTING.md]: Needs to point to licence too

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  "author": {
+    "name": "BBC Psammead Maintainers",
+    "email": "PsammeadMaintainers@bbc.co.uk"
+  },
+```
+
+```
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/BBC-News/psammead/tree/latest/packages/Headings"
```
Comment (dr3) [inline CONTRIBUTING.md]:
```suggestion
    "url": "https://github.com/BBC-News/psammead/tree/latest/packages/psammead-headings"
```

File: CONTRIBUTING.md (hunk 2 above)
Comment (radiocontrolled) [reply CONTRIBUTING.md]: Thanks - done

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+**package.json**
+
+The `package.json` is namespaced with `@bbc/psammead-*` e.g. `@bbc/psammead-headings`.
+
+Include the following authorship and repository information in your package:
+
+```
+  "author": {
+    "name": "BBC Psammead Maintainers",
```
Comment (dr3) [inline CONTRIBUTING.md:38]: What if another bbc team makes their own? we wouldn't be the author right? Probably one for slack

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+| 0.1.0 | [PR#49](https://github.com/BBC-News/psammead/pull/49) Initial creation of package |
+```
+
+
+**README**
+
+The `README.md` should cover the following territory: 
+
+```
+- ## Description (high-level summary that will be accessible to product/UX/Test, plus link to Storybook, explain the different cases)
```
Comment (ChrisBAshton) [inline CONTRIBUTING.md]:
```suggestion
## Description
(high-level summary that will be accessible to product/UX/Test, plus link to Storybook, explain the different cases)
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+- ## When to use this component (including where it is currently used)
```
Comment (ChrisBAshton) [inline CONTRIBUTING.md]:
```suggestion
## When to use this component
(including where it is currently used)
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+- ## When to not use this component (if this is not known, can leave the section empty)
```
Comment (ChrisBAshton) [inline CONTRIBUTING.md]:
```suggestion
## When to not use this component
(if this is not known, can leave the section empty)
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+- ## Accessibility notes (ensuring they make sense to non-technical audiences)
```
Comment (ChrisBAshton) [inline CONTRIBUTING.md]:
```suggestion
## Accessibility notes
(ensuring they make sense to non-technical audiences)
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+- ## Roadmap (what we have planned for this component)
```
Comment (ChrisBAshton) [inline CONTRIBUTING.md]:
```suggestion
## Roadmap
(what we have planned for this component)
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+- ## Additional notes (link to any relevant ADRs)
```
Comment (ChrisBAshton) [inline CONTRIBUTING.md]:
```suggestion
## Additional notes
(link to any relevant ADRs)
```

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+**Tests**
+
+When you run `npm run test` all unit tests should pass and there should be 100% test coverage of the component. No JS or CSS lint errors should be present.
+
+
+**CHANGELOG**
+
+The changelog should follow the following convention: 
+```
+<h1 align="center">PackageName Changelog</h1>
```
Comment (ChrisBAshton) [inline CONTRIBUTING.md]:
```suggestion
# PackageName Changelog
```

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/BBC-News/psammead/tree/latest/packages/psammead-headings"
+  },
+```
+
+
+**Tests**
+
+When you run `npm run test` all unit tests should pass and there should be 100% test coverage of the component. No JS or CSS lint errors should be present.
```
Comment (ChrisBAshton) [inline CONTRIBUTING.md]:
```suggestion
When you run `npm test` all unit tests should pass and there should be 100% test coverage of the component. No JS or CSS lint errors should be present.
```

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+├── index.jsx
+├── index.stories.jsx
+├── index.test.jsx
+├── package.json
+└── package-lock.json
+```
+
+The `__snapshots__` directory is optional.
+
+
```
Comment (ChrisBAshton) [inline CONTRIBUTING.md:29]: delete newline

**Review by ChrisBAshton — CHANGES_REQUESTED** [review]
Good work 👍 main suggestion is to split out the `**README**` section so that people can copy and paste it more easily and it retains good formatting (want to avoid overly long headings, and/or inappropriate list-items creeping into the README)

File: CONTRIBUTING.md (hunk 13 above)
Comment (radiocontrolled) [reply CONTRIBUTING.md:29]: There are two carriage returns before each bold in this readme. I think this is fine

File: CONTRIBUTING.md (hunk 4 above)
Comment (radiocontrolled) [reply CONTRIBUTING.md:38]: @dr3 makes their own what? Can you clarify what you mean please.

Comment (dr3) [reply CONTRIBUTING.md:38]: Makes their own package. If bbc sport make a football table, we wouldnt be the author

Comment (ChrisBAshton) [reply CONTRIBUTING.md:38]: I guess just add a note above this saying "replace team name and email address if appropriate".

File: CONTRIBUTING.md (hunk 13 above)
Comment (ChrisBAshton) [reply CONTRIBUTING.md:29]: So there are! Looks odd to me but agreed either delete them all or leave it 👍

**Review by ChrisBAshton — APPROVED** [review]

**Review by dr3 — APPROVED** [review]

**Pull request merged**